refactor(router): refactor documentation withHashLocation return type

Nicoss54 opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
 This Pull request is a change on the router documentation

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?

Currenty, according to the documentation, withHashLocationStrategy function return RouterConfigurationFeature type


## What is the new behavior?

Now withHashLocationStrategy return the correct alias type:  RouterHashLocationFeature


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

JeanMeche commented:
Hi Nicolas, can you reword your commit message, this is more a `refactor` than a doc change !

Nicoss54 commented:
> Hi Nicolas, can you reword your commit message, this is more a `refactor` than a doc change !

@JeanMeche done, thanks for the feedback

atscott reviewed (approved):
reviewed-for: fw-router, public-api

jessicajaniuk reviewed (approved):
reviewed-for: public-api

JeanMeche commented:
I've pinged Nico about the linting issue. I'll provide some help if needed 👍

JeanMeche reviewed (changes requested):
There are 2 things remaning before we can merge this: 

* fix the linting issue (format) 
* squash your commits

Nicoss54 committed:
refactor(router): refactor documentation withHashLocation return type

This improve the documentation of withHashLocation function by returning the correct type.

atscott commented:
This PR was merged into the repository by commit e25006b341503d7e227d9bcbe1fd2fe66dc8d651.
